pages/index.js

```diff
@@ -4,26 +4,48 @@ import DateDiff from "../components/DateDiff";
 import Overview from "../components/Overview";
 import BodyDimensions from "../components/BodyDimensions";
 import CreateTrainingButton from "../components/CreateTrainingButton";
+import Welcome from "../components/Welcome";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
 
 export default function Home({ setFormData, formData }) {
   const todayDate = new Date();
+  const router = useRouter();
+
+  useEffect(() => {
+    // This check ensures that the router functions are only used on the client
+    if (typeof window !== "undefined") {
+      if (formData.welcome === true) {
+        if (!formData.firstName && !formData.lastName && !formData.firstFill) {
+          router.push("/profile");
+        }
+      }
+    }
+  }, [formData.welcome]);
+
   return (
     <main>
-      <Heading>Hello {formData.firstName},</Heading>
-      <SubHeading>
-        Days:{" "}
-        <DateDiff
-          sDate={formData.entryDate}
-          eDate={todayDate.toLocaleDateString()}
-        />
-      </SubHeading>
-      <Overview weight={formData.weight} />
-      <BodyDimensions
-        chestD={formData.chest}
-        bellyD={formData.belly}
-        hipD={formData.hip}
-      />
-      <CreateTrainingButton />
+      {formData.welcome === true ? (
+        <>
+          <Heading>Hello {formData.firstName},</Heading>
+          <SubHeading>
+            Days:{" "}
+            <DateDiff
+              sDate={formData.entryDate}
+              eDate={todayDate.toLocaleDateString()}
+            />
```

Comment:
Since your ``<DateDiff>`` only makes a calculation and renders nothing, you should not call it a 'component' and instead you can make a ``/lib/utils.js`` - file where you can export named service-functions. 
Nevertheless it seem not to work. In my preview I only see: ``NaN``.